packages/pigeon/lib/swift_generator.dart

```diff
@@ -19,16 +19,21 @@ class SwiftOptions {
   /// Creates a [SwiftOptions] object
   const SwiftOptions({
     this.copyrightHeader,
+    this.errorClassName,
   });
 
   /// A copyright header that will get prepended to generated code.
   final Iterable<String>? copyrightHeader;
 
+  /// The name of the error class used for passing custom error parameters.
+  final String? errorClassName;
+
   /// Creates a [SwiftOptions] from a Map representation where:
   /// `x = SwiftOptions.fromList(x.toMap())`.
   static SwiftOptions fromList(Map<String, Object> map) {
     return SwiftOptions(
       copyrightHeader: map['copyrightHeader'] as Iterable<String>?,
+      errorClassName: map['errorClassName'] as String?,
```

Comment:
is it open for developers to use a custom class name? why don't we just use `PigeonError`?

Reply:
> is it open for developers to use a custom class name? why don't we just use PigeonError?

Generally, I believe calling pigeon once for each package (plugin, app, etc.) should suffice. In these scenarios, there's no need to concern ourselves with the generated class name. However, in this repo, pigeon is invoked multiple times for the same package, which results in multiple classes with the same name, ultimately causing a compilation error.

Kotlin encounters a similar problem but has a solution for it. This [Link](https://github.com/flutter/packages/pull/3234#issuecomment-1445022950) provides a background of the issue as well as its resolution. I have therefore applied the same solution to the generation of Swift code.

I believe overriding the error class name as an advanced usage, so I prefer not to integrate it with the command line option, similar to the approach taken by Kotlin.